### DISCUSSION TITLE
Submitting doesn't return empty values

### DISCUSSION THREAD

**Discussion by jtwebb** [react-hook-form/react-hook-form#9817] [General]
I'm not sure if this is a bug or not, but it seems to be going in direct contrast to https://github.com/react-hook-form/react-hook-form/issues/656. I realize that was several years ago and things change. Is it now normal for empty values to be filtered out? Is there a way I can get all values back when submitting the form?

**Comment by leapful** [comment]
@jtwebb 

Please provide a codesandbox to re-produce the issue.

Based on your issue description, the root cause of submit is stopped might be from validation rules.

**Reply by jtwebb** [reply]
@leapful I think you're right. When I comment out the `yupResolver` I get all the values back. Do you know why this would be happening? Any way to fix it?

https://codesandbox.io/s/awesome-hermann-8refbk?file=/src/App.js

**Reply by leapful** [reply]
@jtwebb 

This is a expected behavior because you're using `onValid` callback of `handleSubmit` method which only be triggered when all validations are valid.

Please use `onError` callback of `handleSubmit` with `getValues` method to get all form values when error occurs on submitting.

Working demo: https://codesandbox.io/s/fragrant-surf-mqt22h?file=/src/App.js

**Reply by jtwebb** [reply]
Thanks for replying! I see that your example works when there are errors, but not when there aren't any errors and the form is valid. I still only get back the filled out form values, not the empty ones (non-required fields).
